hedera-node/hedera-app/src/main/java/com/hedera/node/app/statedumpers/StateDumper.java

```diff
@@ -142,9 +143,15 @@ public static void dumpModChildrenFrom(
                 requireNonNull(state.getChild(state.findNodeIndex(TokenService.NAME, ACCOUNTS_KEY)));
         dumpModAccounts(Paths.get(dumpLoc, SEMANTIC_ACCOUNTS), accounts, checkpoint);
 
-        final VirtualMap<OnDiskKey<ContractID>, OnDiskValue<Bytecode>> contracts = requireNonNull(state.getChild(
+        final VirtualMap<OnDiskKey<ContractID>, OnDiskValue<Bytecode>> byteCodes = requireNonNull(state.getChild(
                 state.findNodeIndex(ContractService.NAME, InitialModServiceContractSchema.BYTECODE_KEY)));
-        dumpModContractBytecodes(Paths.get(dumpLoc, SEMANTIC_CONTRACT_BYTECODES), contracts, checkpoint);
+        dumpModContractBytecodes(
+                Paths.get(dumpLoc, SEMANTIC_CONTRACT_BYTECODES),
+                byteCodes,
+                accounts,
+                (StateMetadata<AccountID, Account>)
+                        state.getServices().get(TokenService.NAME).get(ACCOUNTS_KEY),
```

Comment:
Any reason not to access the service like line ~142? (Then `MerkleHederaState` wouldn't need `getServices()`)
```
state.getChild(state.findNodeIndex(TokenService.NAME, ACCOUNTS_KEY))
```

Reply:
Yeah all we needed was `StateMetadata` and could not see any other way